docs/modules/ROOT/pages/erc1155.adoc

```diff
@@ -1,548 +1,283 @@
+
+    // ISRC5
+    fn supports_interface(self: @TState, interface_id: felt252) -> bool;
+
+    // IERC1155MetadataURI
+    fn uri(self: @TState, token_id: u256) -> ByteArray;
+
+    // IERC1155Camel
+    fn balanceOf(self: @TState, account: ContractAddress, tokenId: u256) -> u256;
+    fn balanceOfBatch(
+        self: @TState, accounts: Span<ContractAddress>, tokenIds: Span<u256>
+    ) -> Span<u256>;
+    fn safeTransferFrom(
+        ref self: TState,
+        from: ContractAddress,
+        to: ContractAddress,
+        tokenId: u256,
+        value: u256,
+        data: Span<felt252>
+    );
+    fn safeBatchTransferFrom(
+        ref self: TState,
+        from: ContractAddress,
+        to: ContractAddress,
+        tokenIds: Span<u256>,
+        values: Span<u256>,
+        data: Span<felt252>
+    );
+    fn isApprovedForAll(self: @TState, owner: ContractAddress, operator: ContractAddress) -> bool;
+    fn setApprovalForAll(ref self: TState, operator: ContractAddress, approved: bool);
 }
 ----
 
 === ERC1155 Compatibility
 
-Although StarkNet is not EVM compatible, this implementation aims to be as close as possible to the ERC1155 standard in the following ways:
-
-* It uses Cairo's `uint256` instead of `felt`.
-* It returns `TRUE` as success.
-
-But some differences can still be found, such as:
+Although Starknet is not EVM compatible, this implementation aims to be as close as possible to the ERC1155 standard 
+using `u256` instead of `felt252`. But some differences can still be found, such as:
 
 * It implements an `uri()` function that returns the same URI for *all* token types. It relies on the token type ID substitution mechanism https://eips.ethereum.org/EIPS/eip-1155#metadata[defined in the EIP]. Clients calling this function must replace the `\{id\}` substring with the actual token type ID.
-* `safeTransferFrom` and `safeBatchTransferFrom` are specified such that the optional `data` argument should be of type bytes.
-In Solidity, this means a dynamically-sized array.
-To be as close as possible to the standard, the `data` argument accepts a dynamic array of felts.
-In Cairo, arrays are expressed with the array length preceding the actual array;
-hence, the method accepts `data_len` and `data` respectively as types `felt` and `felt*`.
-* `IERC1155Receiver` compliant contracts (`ERC1155Holder`) return a hardcoded selector id according to EVM selectors, since selectors are calculated differently in Cairo.
-This is in line with the ERC165 interfaces design choice towards EVM compatibility.
-See the xref:introspection.adoc[Introspection docs] for more info.
-* `IERC1155Receiver` compliant contracts (`ERC1155Holder`) must support ERC165 by registering the `IERC1155Receiver` selector id in its constructor and exposing the `supportsInterface` method.
+* `safe_transfer_from` and `safe_batch_transfer_from` are specified such that the optional `data` argument should be of type `Span<felt252>` simulating bytes.
```

Comment:
I would say we are not, but depends on what we are calling `bytes`. If we are referring to a dynamically sized array, then we are technically simulating that, even when is not an array of u8. I removed the simulating bytes from the sentence because I think is not that important anyway, but to mention that in the ERC721 docs, we are mentioning something similar that maybe we should consider rephrasing.

Reply:
> I would say we are not, but depends on what we are calling bytes. If we are referring to a dynamically sized array, then we are technically simulating that, even when is not an array of u8. I removed the simulating bytes from the sentence because I think is not that important anyway,

Right, I was thinking of the conversation we had regarding if we should change `data`'s type to `ByteArray`. But yeah, it's ultimately not that important

> ...but to mention that in the ERC721 docs, we are mentioning something similar that maybe we should consider rephrasing.

Agreed, good call